book/workflow.rst

```diff
@@ -0,0 +1,236 @@
+
+    workflows:
+        phone_call:
+            # ...
+            transitions:
+                connected:
+                    label: 'Connected'
+                    step_to: start_conversation
+                    transition_definition: connected_definition
+                not_answered:
+                    label: "Not answered"
+                    step_to: end_call
+                    transition_definition: not_answered_definition
+                end_conversation:
+                    label: 'End conversation'
+                    step_to: end_call
+                    transition_definition: end_conversation_definition
+
+Transition Pre-Conditions and Post Actions
+..........................................
+
+.. note::
+
+    You can configure as many workflows as you like, but you can only enable one workflow per
+    entity type at the same time.
+
+.. seealso::
+
+    Read more about all the available options in
+    :doc:`the workflow reference </reference/format/workflow>`.
+
+Applying Workflows to an Entity
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In order to be able to apply a workflow to an entity, the entity must have associations to the
+:class:`Oro\\Bundle\\WorkflowBundle\\Entity\\WorkflowItem` and
+:class:`Oro\\Bundle\\WorkflowBundle\\Entity\\WorkflowStep` classes. The property holding a
+reference to a ``WorkflowItem`` instance keeps the information which workflow is currently being
+applied to the entity. The ``WorkflowStep`` instance is used to store the current state of the
+entity in this workflow:
+
+.. code-block:: php
+    :linenos:
+
+    // src/Acme/DemoBundle/Entity/BlogPost.php
+    namespace Acme\DemoBundle\Entity;
+
+    use Doctrine\ORM\Mapping as ORM;
+    use Oro\Bundle\WorkflowBundle\Entity\WorkflowItem;
+    use Oro\Bundle\WorkflowBundle\Entity\WorkflowStep;
+
+    /**
+     * @ORM\Entity()
+     */
+    class BlogPost
+    {
+        // ...
+
+        /**
+         * @ORM\OneToOne(targetEntity="Oro\Bundle\WorkflowBundle\Entity\WorkflowItem")
+         */
+        private $workflowItem;
+
+        /**
+         * @ORM\OneToOne(targetEntity="Oro\Bundle\WorkflowBundle\Entity\WorkflowStep")
+         */
+        private $workflowStep;
+
+        public function getWorkflowItem()
+        {
+            return $this->workflowItem;
+        }
+
+        public function setWorkflowItem(WorkflowItem $workflowItem)
+        {
+            $this->workflowItem = $workflowItem;
+        }
+
+        public function getWorkflowStep()
+        {
+            return $this->workflowStep;
+        }
+
+        public function setWorkflowStep(WorkflowStep $workflowStep)
+        {
+            $this->workflowStep = $workflowStep;
+        }
+    }
+
+.. tip::
+
+    You do not need to create those methods for extended entities or for entities created in the
+    user interface. The needd properties and methods will be automatically generated when the first
+    workflow is applied to them.
+
+Processes
+---------
```

Comment:
Processes section not finished yet - maybe would be better to introduce all information about processes in scope on another pull request?